modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -235,6 +258,79 @@ private CompletableFuture<Object> processScanRetrieveBatchAction(ReadWriteScanRe
         });
     }
 
+    /**
+     * Process transaction finish request:
+     * <ol>
+     *     <li>Evaluate commit timestamp.</li>
+     *     <li>Run specific raft {@code FinishTxCommand} command, that will apply txn state to corresponding txStateStorage.</li>
+     *     <li>Send cleanup requests to all enlisted primary replicas.</li>
+     * </ol>
+     * This operation is NOT idempotent, because of commit timestamp evaluation.
+     *
+     * @param request Transaction finish request.
+     * @return future result of the operation.
+     */
+    private CompletableFuture<Object> processTxFinishAction(TxFinishReplicaRequest request) {
+        HybridTimestamp commitTimestamp = hybridClock.now();
```

Comment:
Need to check the RAFT term to make sure that primary replicas of enlisted partitions were not changed.

Reply:
Good catch, however, as was discussed, it'll be implemented after IGNITE-17521 re-implementation, or even during one.